src/main/java/net/sf/jabref/external/ExternalFileTypeEntryEditor.java

```diff
@@ -184,10 +181,16 @@ public void changedUpdate(DocumentEvent documentEvent) {
             });
         }
 
+        String title = "Edit file type";
+        
+        if(entry.getName().isEmpty()) {
+            title = "Add new file type";
+        }
+        
         if (dParent != null) {
-            diag = new JDialog(dParent, Localization.lang("Edit file type"), true);
+            diag = new JDialog(dParent, Localization.lang(title), true);
```

Comment:
I'll sort this out. I have a better idea of the translation process now.

The reason for using variables in this way (in general) would be to only need to change it in one position, if you for some reason do want to change it. For translations there is obviously only going to be one string to translate anyway, but for consistency it can make sense to do it like this.

I assume it may be better to do something like

``` Java
private final String editTitle = Localization.lang("Edit file type");
private final String newTitle = Localization.lang("New file type");
...
        String title = editTitle;       
        if(entry.getName().isEmpty()) {
            title = newTitle;
       }
```
and so on?

Regarding @koppor:s comment for lines 215-219, I'll look into the actual operation again. I had the impression that one can use the same dialog if you click on another item, but it may be that it wasn't the case and that I forgot to change back. If that's not the case, I assume

``` Java
        String title = Localization.lang("Edit file type");       
        if(entry.getName().isEmpty()) {
            title = Localization.lang("New file type");
       }
```

is a slightly better solution? (Assuming that you more commonly edit than add file types.)

Will continue working on this PR tonight.